src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -19,14 +19,23 @@ public IntakeSubsystem() {
   /**
    * @return a command that will run forever that sets the motor to intake
    */
+  private Command runIntake(boolean inverted) {
+    return Commands.runEnd(
+        () -> motor.set(IntakeConstants.SPEED * (inverted ? -1 : 1)), () -> motor.set(0), this);
+  }
+
   private Command runIntake() {
-    return Commands.runEnd(() -> motor.set(IntakeConstants.SPEED), () -> motor.set(0), this);
+    return runIntake(false);
   }
 
   public Command intake(ShootSubsystem shooter) {
+    return intake(shooter, false);
+  }
+
+  public Command intake(ShootSubsystem shooter, boolean reversed) {
```

Comment:
This performs a full intake including ensuring the shooter is properly configured, and the note gets pulled into the correct position into the bot. Running the intake in reverse during this process would result in a command that never exits because the note never enters the shooter.

Reply:
Got it. Should I just make it run for a set time, then? Another option is to make it run while the driver is holding the trigger. Also, could you direct me to the line that interrupts when a note is detected?

+we should probably be able to manually kill intake

Reply:
Just run for a set amount of time for now. I want to do some more physical testing of ways the note can get stuck before we make a final decision